### PR TITLE
Booking check out

### PR DESCRIPTION
**Pull request by jaz998**
The changes are primarily about the bookingcheckout.php page. 

This page takes the booking start and end time from booking2.php page and validate again whether the car is already booked before inserting the booking into the database.

Currently the info in the credit card details form on this page is not inserted into the database - this will happen in future commits. 

The codes also do not validate whether the user is logined or send confirmation email - will be done in future commits.

### CODE REVIEW

File: classes/Payment.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          $Address,
+          $State,
+          $PostCode;
+  // Default constructor
+  public function __construct($user = null)
+  {
+    $this->_db = DB::getInstance();
+    $this->_sendMail = Email::getInstance();
+
+    $this->_sessionName = Config::get('sessions/session_name');
```
Comment (s3666874) [inline classes/Payment.php]: @jaz998 the session name is car_buddy. Is that what you want to use? Or have you changed it in another PR?

File: classes/Payment.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    $this->CardNumber= $CardNumber;
+    $this->ExpMonth = $ExpMonth;
+    $this->ExpYear = $ExpYear;
+    $this->CardCVV = $CardCVV;
+    $this->Address = $Address;
+    $this->State = $State;
+    $this->PostCode = $PostCode;
+    
+
+    Session::put($this->_sessionName, $this);
```
Comment (s3666874) [inline classes/Payment.php]: I've had issues with putting the entire object in session. You need to serialize and deserialize it in order to store it properly.

File: lib/2b-lib-res.php [hunk 3]
```diff
@@ -106,13 +106,53 @@ function bookRange ($name, $email, $tel, $start, $end, $notes="", $vehicle_id) {
       return false;
     }
     
-    //$vehicle_id = 6;
+    //Check whether the user has logined
+//    session_start();
+//    if ( !isset($_SESSION["carbuddy"]) || (isset($_SESSION["carbuddy"]) && !$_SESSION["carbuddy"]->isLoggedIn())) {
+//        return 2;
+//    }
+    if(!session_start()){
```
Comment (s3666874) [inline lib/2b-lib-res.php]: I don't think you need to do this. Just calling session_start, PHP will start or resume an existing session automatically for you.

File: classes/Payment.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  // Set private variables
+  private $FullName,
+          $Email,
+          $NameOnCard,
+          $CardNumber,
+          $ExpMonth,
+          $ExpYear,
+          $CardCVV,
+          $Address,
+          $State,
```
Comment (jaz998) [inline classes/Payment.php]: testing comment

**Comment by jaz998** [comment]
@s3666874 , thanks for your comments, I've made changes accordingly, pls have a look, thanks.

**Pull request merged**